readme.md

```diff
@@ -124,6 +124,32 @@ const emitter = require('./some-event-emitter');
 ```
 
 
+##### error
+
+Type: `Function`
+
+A function which checks the value provided by `event` for error conditions.
+If `filter` returns false that takes precedence.
+
+Possible return values:
+* `false`: The event is not an error and will resolve the promise.
+* `true`: The event is an error and will reject the promise using `new Error(value)`.
+* `instanceof Error`: The event is an error and will reject the promise using the
+  return value.
+* Any other return: The event is an error and will reject with a new `Error` using
```

Comment:
I think this is too loose. I would prefer to not allow this. Users can just explicitly wrap the value in an error if they want it.

Reply:
So do you think just support explicit `false` return to not reject, otherwise any other value to be used directly as the Promise rejection value?

Another thought maybe it would be better to not use a return value at all?  Instead of error or shouldReject, checkValue:
```js
{
  checkValue(value) {
    if (value !== 0) { throw new Error('Unexpected exit-code'); }
  }
```

Then again I'm now seeing that this could just be done on the return from pEvent:
```js
return pEvent(/* args */).then(value => {
  if (value !== 0) { throw new Error('Unexpected exit-code'); }
  return value;
})
```

Up to you, I'm ready to drop this request unless you think it's still worth adding.

Reply:
Yeah, let's drop it. It's easy enough to do with `.then()`. Maybe you could the above as a tip to the readme instead?

Reply:
I've added the tip and fixed another issue in the readme.  If you prefer I don't show both await and .then style handling I can remove one of them.